node/interfaces.go

```diff
@@ -30,6 +30,8 @@ type ConsensusEngine interface {
 		blkRequester consensus.BlkRequester, stateResetter consensus.ResetStateBroadcaster, discoveryBroadcaster consensus.DiscoveryReqBroadcaster) error
 
 	CheckTx(ctx context.Context, tx *ktypes.Transaction) error
+
+	ConsensusParams() *ktypes.ConsensusParams
```

Comment:
This is kind awkward, I have to define this on both ConsensusEngine and Node to pass over to chain rpc

Reply:
On this point, I think we are likely to have block executor provide.  OK for now

Reply:
Yeah, we could have one more interface using  BlockProcessor to provide these functions.